### PR TITLE
Clean up tests that had min_pyver=3.8 test setting

### LINKED ISSUE #6306: Cleanup after we drop support for Python 3.7
When we do drop support because 3.7 has reached its end of life after 2023-06-27, we'll need to do some cleanup:

- [x] Remove the code for ``_ExtendAction`` (#8793)
- [x] Import ``Final``, ``Literal``, and ``TypedDict`` from typing directly
- [x] Remove ``functools.cached_property`` import guards

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description
Follow-up to #8792.

Closes #6306

### CODE REVIEW

File: doc/whatsnew/2/2.15/index.rst [hunk 1]
```diff
@@ -383,9 +383,6 @@ Release date: 2022-09-07
 - Fix false positive for ``too-many-function-args`` when a function call is assigned to a class attribute inside the class where the function is defined.
 
   Closes #6592 (`#6592 <https://github.com/pylint-dev/pylint/issues/6592>`_)
-- Fix ``used-before-assignment`` for functions/classes defined in type checking guard.
-
-  Closes #7368 (`#7368 <https://github.com/pylint-dev/pylint/issues/7368>`_)
```
Comment (jacobtylerwalls) [inline doc/whatsnew/2/2.15/index.rst:388]: Other cleanup #1: this was given twice, see above under 2.15.2.

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 2]
```diff
@@ -1133,16 +1133,12 @@ def _check_super_with_arguments(self, node: nodes.Call) -> None:
...
-        # pylint: disable=too-many-boolean-expressions
         if (
             len(node.args) != 2
-            or not isinstance(node.args[1], nodes.Name)
+            or not all(isinstance(arg, nodes.Name) for arg in node.args)
             or node.args[1].name != "self"
-            or not isinstance(node.args[0], nodes.Name)
-            or not isinstance(node.args[1], nodes.Name)
-            or node_frame_class(node) is None
-            # TODO: PY38: Use walrus operator, this will also fix the mypy issue
```
Comment (jacobtylerwalls) [inline pylint/checkers/refactoring/refactoring_checker.py:1144]: Other cleanup 2: resolve this TODO

File: tests/functional/g/generic_alias/generic_alias_side_effects.rc [hunk 3]
```diff
@@ -1,2 +0,0 @@
-[testoptions]
```
Comment (Pierre-Sassoulas) [inline tests/functional/g/generic_alias/generic_alias_side_effects.rc:1]:
```suggestion
py-version=3.8
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
Look great already ! Shouldn't we add py-version instead of removing rc file when the py-version is not already set ?

File: tests/functional/g/generic_alias/generic_alias_side_effects.rc (hunk 3 above)
Comment (jacobtylerwalls) [reply tests/functional/g/generic_alias/generic_alias_side_effects.rc:1]: I see the benefit of doing that when removing the `max_pyver=3.8` setting, and that was done in #8609. But what would be the benefit of doing it when removing `min_pyver`? When testing on 3.8, there is no difference, and when testing on 3.9, we would be saying "test as if on 3.8", but again, no difference?

Comment (jacobtylerwalls) [reply tests/functional/g/generic_alias/generic_alias_side_effects.rc:1]: I found this to be a real brain-teaser :-}

Comment (Pierre-Sassoulas) [reply tests/functional/g/generic_alias/generic_alias_side_effects.rc:1]: I thought again about it, unless we're testing something specific to python 3.8 (like the need to have from __future__ import annotations``) and if we're not modifying the test itself then the py-version is useless in the conf.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Need a rebase for the primer to work ?

**Pull request merged**